src/main/java/uk/gov/hmcts/darts/task/runner/impl/AudioLinkingAutomatedTask.java

```diff
@@ -0,0 +1,108 @@
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+import uk.gov.hmcts.darts.event.enums.EventStatus;
+import uk.gov.hmcts.darts.log.api.LogApi;
+import uk.gov.hmcts.darts.task.api.AutomatedTaskName;
+import uk.gov.hmcts.darts.task.config.AutomatedTaskConfigurationProperties;
+import uk.gov.hmcts.darts.task.runner.AutoloadingManualTask;
+import uk.gov.hmcts.darts.task.service.LockService;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+@Component
+public class AudioLinkingAutomatedTask extends AbstractLockableAutomatedTask
+    implements AutoloadingManualTask {
+
+    private final EventRepository eventRepository;
+    private final MediaRepository mediaRepository;
+    private final MediaLinkedCaseRepository mediaLinkedCaseRepository;
+
+    @Getter
+    private final Integer audioBufferSeconds;
+    private final HearingRepository hearingRepository;
+
+    protected AudioLinkingAutomatedTask(AutomatedTaskRepository automatedTaskRepository,
+                                        AutomatedTaskConfigurationProperties automatedTaskConfigurationProperties,
+                                        LogApi logApi, LockService lockService,
+                                        EventRepository eventRepository,
+                                        MediaRepository mediaRepository, MediaLinkedCaseRepository mediaLinkedCaseRepository,
+                                        @Value("${darts.automated-tasks.audio-linking.audio-buffer-seconds:0}")
+                                        Integer audioBufferSeconds, HearingRepository hearingRepository) {
+        super(automatedTaskRepository, automatedTaskConfigurationProperties, logApi, lockService);
+        this.eventRepository = eventRepository;
+        this.mediaRepository = mediaRepository;
+        this.mediaLinkedCaseRepository = mediaLinkedCaseRepository;
+        this.audioBufferSeconds = audioBufferSeconds;
+        this.hearingRepository = hearingRepository;
+    }
+
+    @Override
+    public AutomatedTaskName getAutomatedTaskName() {
+        return AutomatedTaskName.AUDIO_LINKING_TASK_NAME;
+    }
+
+    @Override
+    @Transactional
+    public void run() {
+        super.run();
+    }
+
+
+    @Override
+    protected void runTask() {
+        List<EventEntity> events = eventRepository.findAllByEventStatus(EventStatus.AUDIO_LINK_NOT_DONE_MODERNISED.getStatusNumber(),
+                                                                        Limit.of(getAutomatedTaskBatchSize()));
+        final Set<MediaLinkedCaseEntity> mediaLinkedCaseEntities = new HashSet<>();
+        final Set<HearingEntity> editedHearingEntities = new HashSet<>();
+
+        events.forEach(event -> processEvent(event, editedHearingEntities, mediaLinkedCaseEntities));
+
+        hearingRepository.saveAll(editedHearingEntities);
+        mediaLinkedCaseRepository.saveAll(mediaLinkedCaseEntities);
+        eventRepository.saveAll(events);
```

Comment:
i had my concerns about this, so spoke to Sashi, and he also agrees that we should nto be saving up 7500 changes and doing them at the end, but processing 1 event at a time, saving all required changes for it, then moving to the next event.

Reply:
saveAll is more performant then individual saves even at large scales

Reply:
its no the performance we are worried about here, its to make sure that 1 failure doesn't bring down the whole batch.
in fact, we should we catching and logging any exception we hit when processing an event, but still moving on to the next event without failing the entire batch.

Reply:
Aha okay :)

Sounds like a hybrid model can be used then so we still use saveAll for the performance gains but we increase error detection to prevent a global failure from a single event

Reply:
Updated :)